### ISSUE TITLE
Ping-pong causes flood upon application restart

### ISSUE THREAD

**Issue by asc-adean** [miguelgrinberg/Flask-SocketIO#1165]
**Describe the bug**
I have the following setup to connect to a socket event in Flask SocketIO:

```
var ping_pong_times = [];
        var start_time;
        window.setInterval(function () {
            start_time = (new Date).getTime();
            player_status_socket.emit('my_ping');
        }, 5000);

        player_status_socket.on('my_pong', function () {
            var latency = (new Date).getTime() - start_time;
            ping_pong_times.push(latency);
            ping_pong_times = ping_pong_times.slice(-30); // keep last 30 samples
            var sum = 0;
            for (var i = 0; i < ping_pong_times.length; i++)
                sum += ping_pong_times[i];
            $('#ping-pong').text(Math.round(10 * sum / ping_pong_times.length) / 10);
        });
```

In flask app:
```
@socketio.on('my_ping', namespace='/pstatus')
def ping_pong():
    socketio.emit('my_pong', namespace='/pstatus')
    socketio.sleep(0)

...

if __name__ == '__main__':
    socketio.run(app, host="0.0.0.0")
```

**To Reproduce**
1. Turn on flask app (`python ws_api.py`)
2. Connect to the route, load index.html
3. Shut down flask app
4. Wait a few minutes
5. Turn app back on (same as step 1)
6. Watch massive flood of ping pongs come through, if you wait longer in Step 4 you may get this error
```
926e8b9fb46b4d688c308e479f48aa87: Sending packet MESSAGE data 2/pstatus,["my_pong",null]
926e8b9fb46b4d688c308e479f48aa87: Received packet MESSAGE data 2/pstatus,["my_ping"]
received event "my_ping" from 926e8b9fb46b4d688c308e479f48aa87 [/pstatus]
emitting event "my_pong" to all [/pstatus]
926e8b9fb46b4d688c308e479f48aa87: Sending packet MESSAGE data 2/pstatus,["my_pong",null]
926e8b9fb46b4d688c308e479f48aa87: Received packet MESSAGE data 2/pstatus,["my_ping"]
received event "my_ping" from 926e8b9fb46b4d688c308e479f48aa87 [/pstatus]
emitting event "my_pong" to all [/pstatus]
926e8b9fb46b4d688c308e479f48aa87: Sending packet MESSAGE data 2/pstatus,["my_pong",null]
926e8b9fb46b4d688c308e479f48aa87: Received packet PING data None
926e8b9fb46b4d688c308e479f48aa87: Sending packet PONG data None
926e8b9fb46b4d688c308e479f48aa87: Received packet MESSAGE data 2/pstatus,["my_ping"]
received event "my_ping" from 926e8b9fb46b4d688c308e479f48aa87 [/pstatus]
Traceback (most recent call last):
  File "/git/indomita-realtime/venv/lib/python3.6/site-packages/engineio/server.py", line 394, in handle_request
    socket.handle_post_request(environ)
  File "/git/indomita-realtime/venv/lib/python3.6/site-packages/engineio/socket.py", line 123, in handle_post_request
    p = payload.Payload(encoded_payload=body)
  File "/git/indomita-realtime/venv/lib/python3.6/site-packages/engineio/payload.py", line 15, in __init__
    self.decode(encoded_payload)
  File "/git/indomita-realtime/venv/lib/python3.6/site-packages/engineio/payload.py", line 76, in decode
    raise ValueError('Too many packets in payload')
ValueError: Too many packets in payload
```

**Expected behavior**
No ping pongs unless I'm actively connected to the socket?

**Comment by miguelgrinberg** [comment]
This is a feature designed to prevent attacks from malicious clients. The default is to reject any payloads that have more than 16 packets. If you need to support more, you can change how many are accepted:

```python
from engineio.payload import Payload
payload.max_decode_packets = 100
```

Though in your case I believe aborting with the "too many packets" error is actually the right behavior, since handling all those pings all at the same time makes no sense.

A better design for your application would be for clients to not send pings when they are disconnected from the server.

**Comment by asc-adean** [comment]
Hi Miguel,

I also wanted to say thank you for this project, it's what got me started using python in general and I've been able to add it to my repertoire of scripting languages on a professional level.

The error is the right behavior, I just needed to design the frontend JS to only fire off when the socket is alive.  I think I was more or less surprised to know those pings were accumulating in the browser and 'released' when the socket came back, rather than them failing due to a nonexistent connection.  I think I've remedied the situation now with:

```
if(player_status_socket.readState === WebSocket.OPEN) {
  --ping pong javascript here--
}
